gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowTriggerHandler.java

```diff
@@ -150,7 +150,8 @@ private boolean persistFlowAction(MultiActiveLeaseArbiter.LeaseObtainedStatus le
     if (this.dagActionStore.isPresent() && this.multiActiveLeaseArbiter.isPresent()) {
       try {
         DagActionStore.DagAction flowAction = leaseStatus.getFlowAction();
-        this.dagActionStore.get().addDagAction(flowAction.getFlowGroup(), flowAction.getFlowName(), flowAction.getFlowExecutionId(), flowAction.getFlowActionType());
+        // Replace flow execution id with trigger timestamp to easily track the flow
+        this.dagActionStore.get().addDagAction(flowAction.getFlowGroup(), flowAction.getFlowName(), String.valueOf(leaseStatus.getEventTimestamp()), flowAction.getFlowActionType());
```

Comment:
seems confusing that `LeaseObtainedStatus.getFlowAction().getFlowExecutionId()` returns a different value than what we're storing in the `DagAction.getFlowExecutionId()`.  e.g. why not unify them by *encapsulating* the switcheroo to `LeaseObtainedStatus.getEventTimestamp()` inside the `LeaseObtainedStatus.getFlowAction()` call?

then, in that location I suggest a code comment explaining why it's helpful and safe to replace the original value w/ this one.